entry_types/scrolled/package/spec/frontend/MediaPlayer-spec.js

```diff
@@ -30,41 +32,103 @@ describe('MediaPlayer', () => {
-    rerender(<MediaPlayer type={'video'} sources={getVideoSources()} />);
+    rerender(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
     expect(spyMedia).toHaveBeenCalledTimes(1);
   });
 
   it('requests new Player on sources change', () => {
     const spyMedia = jest.spyOn(media, 'getPlayer');
-    const {rerender} = render(<MediaPlayer type={'video'} sources={getVideoSources()} />);
+    let state = getInitialPlayerState();
+    let playerActions = getPlayerActions();
+    const {rerender} = render(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
     expect(spyMedia).toHaveBeenCalledTimes(1);
     let videoSources = getVideoSources();
     videoSources[0].src = 'example.mp4';
-    rerender(<MediaPlayer type={'video'} sources={videoSources} />);
+    rerender(<MediaPlayer type={'video'} sources={videoSources} playerState={state} playerActions={playerActions} />);
     expect(spyMedia).toHaveBeenCalledTimes(2);
   });
 
-  it('releases media player on component unmount', () => {
-    const {container, unmount} = render(<MediaPlayer type={'audio'} sources={getAudioSources()} />);
-    const spyMediaRelease = jest.spyOn(media, 'releasePlayer');
-    unmount();
-    expect(spyMediaRelease).toHaveBeenCalled();
-    expect(container.querySelector('audio')).toBeNull();
+  it('calls play on player when shouldPlay changes to true in playerState', () => {
+    let state = getInitialPlayerState();
+    let playerActions = getPlayerActions();
+    const {rerender} = render(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
+    
+    state = getInitialPlayerState();
+    state.shouldPlay = true;
+
+    let spyPlay = jest.spyOn(media.players['0'], 'play');
+    rerender(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
+    expect(spyPlay).toHaveBeenCalledTimes(1);
   });
 
+  // postpone till volumebinding is added to videoPlayer
+  // it('calls playAndFadeIn on player when shouldPlay changes to true and fadeDuration is present', () => {
+  //   let state = getInitialPlayerState();
+  //   let playerActions = getPlayerActions();
+  //   const {rerender} = render(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
+    
+  //   state = getInitialPlayerState();
+  //   state.fadeDuration = 100;
+  //   state.shouldPlay = true;
+  //   console.log(media.players['0'].playAndFadeIn);
+    
+  //   let spyPlayAndFadeIn = jest.spyOn(media.players['0'], 'playAndFadeIn');
+  //   rerender(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
+  //   expect(spyPlayAndFadeIn).toHaveBeenCalledTimes(1);
+  //   expect(spyPlayAndFadeIn).toHaveBeenCalledWith(100);
+  // });
```

Comment:
This spec passes if we enable volume fading [here](https://github.com/codevise/pageflow/pull/1412/files#diff-1763f9021578071122c83cf4dbaef00fL30).

Reply:
sorry where? it is navigating to the changes tab I did not get what you mean

Reply:
Sorry for the broken link. I mean here:

https://github.com/codevise/pageflow/blob/43ca01f51d1dd2ab150f0ec2e692035da81ae7be/package/src/frontend/media/createMediaPlayer.js#L30